server/app/models/Version.java

```diff
@@ -171,4 +179,29 @@ public boolean removeTombstoneForQuestion(Question question) {
     }
     return this.tombstonedQuestionNames.remove(question.getQuestionDefinition().getName());
   }
+
+  @VisibleForTesting
+  public boolean addTombstoneForProgramForTest(Program program) {
+    if (this.tombstonedProgramNames == null) {
+      this.tombstonedProgramNames = new ArrayList<>();
+    }
+    if (this.programIsTombstoned(program.getProgramDefinition().adminName())) {
+      return false;
+    }
+    return this.tombstonedProgramNames.add(program.getProgramDefinition().adminName());
+  }
+
+  public boolean removeTombstoneForProgram(Program program) {
```

Comment:
suggestion (blocking): please document this method

Reply:
Done.